libs/utils/executor.py

```diff
@@ -130,19 +134,28 @@ def __init__(self, target_conf=None, tests_conf=None):
     def run(self):
         self._print_section('Executor', 'Experiments execution')
 
+        self.experiments = []
+
         # Run all the configured experiments
-        exp_idx = 1
         for tc in self._tests_conf['confs']:
             # TARGET: configuration
             if not self._target_configure(tc):
                 continue
             for wl_idx in self._tests_conf['wloads']:
                 # TEST: configuration
-                wload = self._wload_init(tc, wl_idx)
+                wload, test_dir = self._wload_init(tc, wl_idx)
                 for itr_idx in range(1, self._iterations + 1):
-                    # WORKLOAD: execution
-                    self._wload_run(exp_idx, tc, wl_idx, wload, itr_idx)
-                    exp_idx += 1
+                    exp = Experiment(
+                        wload_name=wl_idx,
+                        wload=wload,
+                        conf=tc,
+                        iteration=itr_idx,
+                        out_dir=os.path.join(test_dir, str(itr_idx)))
+                    self.experiments.append(exp)
+
+            # WORKLOAD: execution
+            for exp_idx, experiment in enumerate(self.experiments):
```

Comment:
I didn't want to do this because it doesn't actually tell you anything useful about the experiment, we only use it to show execution progress in the log output. I think we could add it in if necessary later, but for now it would bloat the data IMO.